chia/timelord/timelord_api.py

```diff
@@ -99,13 +101,22 @@ async def new_peak_timelord(self, new_peak: timelord_protocol.NewPeakTimelord) -
             self.timelord.state_changed("skipping_peak", {"height": new_peak.reward_chain_block.height})
 
     def check_orphaned_unfinished_block(self, new_peak: NewPeakTimelord):
+        new_peak_unf_rh = new_peak.reward_chain_block.get_unfinished().get_hash()
         for unf_block in self.timelord.unfinished_blocks:
             if unf_block.reward_chain_block.total_iters <= new_peak.reward_chain_block.total_iters:
+                if unf_block.reward_chain_block.get_hash() == new_peak_unf_rh:
```

Comment:
we would expect this unfinished block to be removed later, when `_handle_new_peak()` is called, right?

Reply:
it is, the test also make sure this happens